sunpy/instr/tests/test_goes.py

```diff
@@ -81,100 +87,348 @@ def test_temp_em():
+
+    # Test case 2: GOESLightCurve object with flux and temperature
+    # data, but no EM data.
+    goes_test = goes.radiative_loss_rate(goeslc_no_em)
+    assert_frame_equal(goeslc_test.data, goeslc_expected.data)
+
+def test_calc_rad_loss_errors():
+    # Define input variables
+    temp = Quantity([11.0, 11.0, 11.0, 11.0, 11.0, 11.0], unit="MK")
+    em = Quantity([4.0e+48, 4.0e+48, 4.0e+48, 4.0e+48, 4.0e+48, 4.0e+48],
+                  unit="1/cm**3")
+    obstime = np.array([datetime.datetime(2014, 1, 1, 0, 0, 0),
+                        datetime.datetime(2014, 1, 1, 0, 0, 2),
+                        datetime.datetime(2014, 1, 1, 0, 0, 4),
+                        datetime.datetime(2014, 1, 1, 0, 0, 6),
+                        datetime.datetime(2014, 1, 1, 0, 0, 8),
+                        datetime.datetime(2014, 1, 1, 0, 0, 10)], dtype=object)
+    temp_toolong = Quantity(np.append(temp.value, 0), unit="MK")
+    obstime_toolong =  np.array([datetime.datetime(2014, 1, 1, 0, 0, 0),
+                        datetime.datetime(2014, 1, 1, 0, 0, 2),
+                        datetime.datetime(2014, 1, 1, 0, 0, 4),
+                        datetime.datetime(2014, 1, 1, 0, 0, 6),
+                        datetime.datetime(2014, 1, 1, 0, 0, 8),
+                        datetime.datetime(2014, 1, 1, 0, 0, 10),
+                        datetime.datetime(2014, 1, 1, 0, 0, 12)], dtype=object)
+    obstime_nonchrono = copy.deepcopy(obstime)
+    obstime_nonchrono[1] = obstime[-1]
+    obstime_nonchrono[-1] = obstime[1]
+    obstime_notdatetime = copy.deepcopy(obstime)
+    obstime_notdatetime[0] = 1
+    temp_outofrange = Quantity([101, 11.0, 11.0, 11.0, 11.0, 11.0], unit="MK")
+    # Ensure correct exceptions are raised.
+    with pytest.raises(ValueError):
+        rad_loss_test = goes.calc_rad_loss(temp_toolong, em, obstime)
+    with pytest.raises(ValueError):
+        rad_loss_test = goes.calc_rad_loss(temp_outofrange, em, obstime)
+    with pytest.raises(IOError):
+        rad_loss_test = goes.calc_rad_loss(temp, em, obstime_toolong)
+    with pytest.raises(TypeError):
+        lx_test = goes.calc_rad_loss(temp, em, obstime_notdatetime)
+    with pytest.raises(ValueError):
+        rad_loss_test = goes.calc_rad_loss(temp, em, obstime_nonchrono)
+
+def test_calc_rad_loss_nokwags():
+    # Define input variables
+    temp = Quantity([11.0, 11.0, 11.0, 11.0, 11.0, 11.0], unit="MK")
+    em = Quantity([4.0e+48, 4.0e+48, 4.0e+48, 4.0e+48, 4.0e+48, 4.0e+48],
+                  unit="1/cm**3")
+    obstime = np.array([datetime.datetime(2014, 1, 1, 0, 0, 0),
+                        datetime.datetime(2014, 1, 1, 0, 0, 2),
+                        datetime.datetime(2014, 1, 1, 0, 0, 4),
+                        datetime.datetime(2014, 1, 1, 0, 0, 6),
+                        datetime.datetime(2014, 1, 1, 0, 0, 8),
+                        datetime.datetime(2014, 1, 1, 0, 0, 10)], dtype=object)
```

Comment:
same than above... in fact... the same is repeated once and again, could not be put outside the functions and used in all?